spikes/api/main.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	mm "github.com/mattermost/platform/model"
+)
+/*
+main -u <username> -p <password> <server URL> [team name]
+Authenticates your login information, then gives you your AuthToken.
+If the team name is unentered or invalid main shows valid team names. 
+*/
+func main() {
+	username := flag.String("u", "", "Username")
+	password := flag.String("p", "", "Password")
+	flag.Parse()
+	url := flag.Arg(0)
+	teamName := flag.Arg(1)
+	client := mm.NewClient(url)
+	_, err := client.Login(*username, *password)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+	
+	fmt.Println("Auth successful! Token: ", client.AuthToken)
+	
+	//Gathers all availible teams in a map, 
+	clientResult, clientAppError := client.GetAllTeamListings()
+	teamMap := clientResult.Data.(map[string]*mm.Team)
+	if clientAppError != nil {
+		fmt.Println(clientAppError)
+		return
+	}
+	//Validates input team name
+	teamObjMap, teamError := client.GetTeamByName(teamName)
```

Comment:
So, I can see that this uses the command-line argument to fetch a team name, but could we try using the team names from the `teamMap` variable? That way, if no team was provided as a command-line argument, we could just use all team names and list all of the channels?